File: sdks/php/src/Resource/Rcs.php
<?php

declare(strict_types=1);

namespace Pushfy\Resource;

use Pushfy\Pushfy;

/** RCS rich messaging. */
final class Rcs
{
    /** @var Pushfy */
    private $client;

    public function __construct(Pushfy $client)
    {
        $this->client = $client;
    }

    /**
     * Send an RCS rich card via the API RCS campaign.
     *
     * @param array{to: string, text: string, title?: string, url?: string, cta?: string, image?: string, extId?: string} $params
     * @return mixed
     */
    public function send(array $params)
    {
        $msg = [
            'destinations' => [['to' => Message::digits($params['to'] ?? '')]],
            'text' => $params['text'] ?? '',
        ];
        foreach (['title' => 'title', 'image' => 'image', 'url' => 'url', 'cta' => 'cta'] as $in => $out) {
            if (isset($params[$in]) && $params[$in] !== null) {
                $msg[$out] = $params[$in];
            }
        }
        if (isset($params['extId']) && $params['extId'] !== null) {
            $msg['ext_id'] = $params['extId'];
        }

        return $this->client->classicRequest('POST', '/apircsnativo.php', [
            'json' => ['messages' => [$msg]],
        ]);
    }
}

File: sdks/go/rcs.go
package pushfy

import "context"

// RCSResource sends RCS rich cards via the API RCS campaign.
type RCSResource struct{ c *Client }

// RCSMessage is a single RCS rich card. Optional fields are omitted when empty.
type RCSMessage struct {
	To    string
	Title string
	Text  string
	URL   string
	CTA   string
	Image string
	ExtID string
}

// Send sends one RCS rich card and returns the accepted messages.
func (r *RCSResource) Send(ctx context.Context, m RCSMessage) ([]MessageResult, error) {
	msg := apiMessage{
		Destinations: []apiDest{{To: onlyDigits(m.To)}},
		Text:         m.Text,
		Title:        m.Title,
		Image:        m.Image,
		URL:          m.URL,
		CTA:          m.CTA,
		ExtID:        m.ExtID,
	}
	var out []MessageResult
	err := r.c.classic(ctx, "POST", "/apircsnativo.php", classicOpts{json: messagesEnvelope{Messages: []apiMessage{msg}}}, &out)
	return out, err
}
